ccf_transaction_processor/README.md

```diff
@@ -18,18 +18,18 @@ TP. Make sure the following env variables are defined:
 
 2) PDO_INSTALL_ROOT : CCF will be installed at PDO_INSTALL_ROOT/opt/pdo/ccf/.
 
-3) PDO_HOME : `export PDO_HOME=${PDO_INSTALL_ROOT}/opt/pdo/`.
+3) HOSTNAME : CCF's first node will be deployed at HOSTNAME:6600. One can simply set HOSTNAME to be the `ip-address` of the VM that can be used to ping the VM from other machines. For local testing, set HOSTNAME to `127.0.0.1`.
```

Comment:
Thank you for the questions. Two things 1) the cchost.toml template that gets expanded (using the same expand_config that pdo uses) will look for HOSTNAME env variable, and if present use that to determine the rpc-address. So the use of HOSTNAME is really because we can use the same expand_config without any change. 2) yes, after the template is expanded, and the config is created, the operator can always go and change the rpc-address in the toml (much like what one can do with pdo eservce.toml), before deploying ccf. The deployment always uses the rpc-address value from the toml file irrespective of what HOSTNAME is. The overall procedure is exactly same as what we do for pdo eservices.

The default for HOSTNAME is "localhost" that gets mapped to 127.0.0.1 in expand_config. But in case, the CCF needs to deployed on a stand alone VM such that it is accessible via other VMs, setting HOSTNAME to fqdns does not seem to really work (the socket.gethostbyname method inside expand config returns an ip that is not very useful), so I wrote in the README to set HOSTNAME to ip-address itself. 

Finally, CCF supports fqdns in addtion to ip-address, the config formats are a bit different, i need to understand this, the best thing will be to use fqdns itself rather than an ip.

Reply:
> he cchost.toml template that gets expanded (using the same expand_config that pdo uses) will look for HOSTNAME env variable, and if present use that to determine the rpc-address

Ah, figured out what the issue is: HOSTNAME is a pre-defined shell variable _but_ is not defined as a _environment variable_.  I guess that's why it was ignored.   Given that already PDO uses `HOSTNAME`  i guess the train has left the station, but it is a bit an unfortunate choice of name ...

Reply:
there is a difference... PDO *uses* HOSTNAME but does not define it. i think we could improve the wording of the documentation. specifically... you should NOT define HOSTNAME.

Reply:
I have no problem relying in predefined `HOSTNAME` if that works, but in this case the usage later stipulates that it should be set 127.0.0.1, at least if you use the default config?

Reply:
added issued #296 to replace the documentation.